composer/callbacks/checkpoint_saver.py

```diff
@@ -210,94 +291,129 @@ class CheckpointSaver(Callback):
-        del logger  # unused
         if self.save_interval(state, Event.EPOCH_CHECKPOINT):
-            self._save_checkpoint(state)
+            log_level = LogLevel.EPOCH if state.get_elapsed_duration() < 1.0 else LogLevel.FIT
+            self._save_checkpoint(state, logger, log_level)
 
-    def _save_checkpoint(self, state: State):
-        checkpoint_filepath_format = os.path.join(self.checkpoint_folder, self.name_format)
+    def _save_checkpoint(self, state: State, logger: Logger, log_level: LogLevel):
+        checkpoint_filepath = os.path.join(format_name_with_dist(self.folder, logger.run_name), self.filename)
         checkpoint_filepaths = checkpoint.save_checkpoint(state,
-                                                          checkpoint_filepath_format,
+                                                          logger,
+                                                          checkpoint_filepath,
                                                           weights_only=self.weights_only)
 
-        if self.save_latest_format is not None and dist.get_global_rank() < len(checkpoint_filepaths):
+        if dist.get_global_rank() < len(checkpoint_filepaths):
+            # Log the checkpoint as an artifact
             checkpoint_filepath = checkpoint_filepaths[dist.get_global_rank()]
-            symlink_name = os.path.join(self.checkpoint_folder, checkpoint.format_name(self.save_latest_format, state))
-            os.makedirs(os.path.dirname(symlink_name), exist_ok=True)
-            try:
-                os.remove(symlink_name)
-            except FileNotFoundError:
-                pass
-            os.symlink(checkpoint_filepath, symlink_name)
+            artifact_name = format_name_with_dist_and_time(self.artifact_name, logger.run_name,
+                                                           state.timer.get_timestamp()).lstrip("/")
+            if state.is_model_deepspeed and not is_tar(artifact_name):
+                # Deepspeed requires tarballs; appending `.tar`
+                artifact_name += ".tar"
+            logger.file_artifact(log_level=log_level,
+                                 artifact_name=artifact_name,
+                                 file_path=checkpoint_filepath,
+                                 overwrite=self.overwrite)
+
+            if self.latest_filename is not None:
+                symlink_name = os.path.join(
+                    format_name_with_dist(self.folder, logger.run_name),
+                    format_name_with_dist_and_time(self.latest_filename, logger.run_name, state.timer.get_timestamp()),
+                )
+                symlink_dirname = os.path.dirname(symlink_name)
+                if symlink_dirname:
+                    os.makedirs(symlink_dirname, exist_ok=True)
+                try:
+                    os.remove(symlink_name)
+                except FileNotFoundError:
+                    pass
+                os.symlink(checkpoint_filepath, symlink_name)
```

Comment:
We can! It will require us to manually implement symlinks on object stores via text files, since object stores do not natively support symlinks. This will be coming in a separate PR :) .... unless if you want this PR to be even bigger haha!